File: api/logcenter/logcenter.go
package logcenter

import (
	"net/http"
	"strconv"

	internalLog "github.com/0xJacky/Nginx-UI/internal/logcenter"
	"github.com/gin-gonic/gin"
)

// GetSources returns all queryable log sources.
func GetSources(c *gin.Context) {
	sources := internalLog.GetSources()
	c.JSON(http.StatusOK, sources)
}

// QueryLogs searches logs from a source.
func QueryLogs(c *gin.Context) {
	source := c.DefaultQuery("source", "nginx_access")
	keyword := c.DefaultQuery("keyword", "")
	limitStr := c.DefaultQuery("limit", "200")
	limit, _ := strconv.Atoi(limitStr)

	lines, err := internalLog.QueryLogs(source, keyword, limit)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}

	c.JSON(http.StatusOK, gin.H{
		"source":  source,
		"keyword": keyword,
		"count":   len(lines),
		"lines":   lines,
	})
}

File: app/src/api/logcenter.ts
import { http } from '@uozi-admin/request'

export interface LogSource {
  id: string
  name: string
  type: string
  description: string
}

export interface LogQueryResult {
  source: string
  keyword: string
  count: number
  lines: string[]
}

export const logcenterApi = {
  getSources: () => http.get<LogSource[]>('/logcenter/sources'),
  queryLogs: (source: string, keyword = '', limit = 200) =>
    http.get<LogQueryResult>(`/logcenter/query?source=${source}&keyword=${encodeURIComponent(keyword)}&limit=${limit}`),
}

export default logcenterApi
